In [1]:
from utils import styled_print, create_dir, extract_images

In [2]:
book_file_path = "./data/fire-and-blood.docx"

## Extract Raw Data From the Docx File

Using Microsoft Word's Word Counter Tool, we found the following information:

| Property      | Count |
| ----------- | ----------- |
| Pages       | 703         |
| Words       | 256,032     |
| Paragraphs  | 3197        |
| Lines       | 23,849      |
| Graphics    | 71          |
| Characters (No Space) | 1,189,352 |
| Characters (With Space) | 1,441,453 |

We have used Microsoft Word's Advance Search Feature (Use flag `^g`) to find total number of graphical objects in our document.

The goal of this part of code is to extract similar amount of data from our document. 

**Note** Our numbers might not exactly match these statics but we will try to capture numbers as close to these stats. 

### Extract Images

In [3]:
image_dir = create_dir("./data", "images")
image_file_paths = extract_images(book_file_path, image_dir, verbose=0)
styled_print(f"Found Total {len(image_file_paths)} Images from the Book {book_file_path}", header=True)

› creating directory ... ./data/images
› Extracting Images from ./data/fire-and-blood.docx
› Found Total 112 Images from the Book ./data/fire-and-blood.docx


**Observations**
- As indicated here, we were able to find total 112 images from the book. 
- We have manually checked all the images to verify that there is no `Non-Image` object present. 
- We have found that our approach to extract images actually `extracted all the correct images` and it is much better that using Microsoft Word's Advance Search option to find images. 